tests/python/rest_api/utils.py

```diff
@@ -41,34 +43,46 @@ def initialize_export(endpoint: Endpoint, *, expect_forbidden: bool = False, **k
     return rq_id
 
 
-def wait_and_download_v2(
+def wait_background_request(
     api_client: ApiClient,
     rq_id: str,
     *,
     max_retries: int = 50,
     interval: float = 0.1,
-    download_result: bool = True,
-) -> Optional[bytes]:
+) -> tuple[models.Request, HTTPResponse]:
     for _ in range(max_retries):
         (background_request, response) = api_client.requests_api.retrieve(rq_id)
         assert response.status == HTTPStatus.OK
         if (
             background_request.status.value
             == models.RequestStatus.allowed_values[("value",)]["FINISHED"]
         ):
-            break
+            return background_request, response
         sleep(interval)
-    else:
-        assert False, (
-            f"Export process was not finished within allowed time ({interval * max_retries}, sec). "
-            + f"Last status was: {background_request.status.value}"
-        )
+
+    assert False, (
+        f"Export process was not finished within allowed time ({interval * max_retries}, sec). "
+        + f"Last status was: {background_request.status.value}"
+    )
+
+
+def wait_and_download_v2(
+    api_client: ApiClient,
+    rq_id: str,
+    *,
+    max_retries: int = 50,
+    interval: float = 0.1,
+    download_result: bool = True,
+) -> Optional[bytes]:
+    background_request, _ = wait_background_request(
+        api_client, rq_id, max_retries=max_retries, interval=interval
+    )
 
     if not download_result:
         return None
 
     # return downloaded file in case of local downloading or None otherwise
-    if background_request.result_url:
+    if download_result and background_request.result_url:
```

Comment:
```suggestion
    if download_result:
        assert background_request.result_url
```

Seems like this would be better for catching mistakes.